### ISSUE TITLE
Bad suggestion: `nonminimal_bool` suggests replacing `!cfg!(windows)` with `true`.

### ISSUE THREAD

**Issue by kennytm** [rust-lang-nursery/rust-clippy#1885]
Test case:

```rust
#![warn(nonminimal_bool)]
fn main() {
    let _ = !cfg!(windows);
}
```

Clippy will warn:

```
warning: this boolean expression can be simplified
 --> src/main.rs:3:13
  |
3 |     let _ = !cfg!(windows);
  |             ^^^^^^^^^^^^^^ help: try `true`
  |
note: lint level defined here
 --> src/main.rs:1:9
  |
1 | #![warn(nonminimal_bool)]
  |         ^^^^^^^^^^^^^^^
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#nonminimal_bool
```

The suggestion (`` try `true` ``) is wrong because the code will lose portability. Instead Clippy should suggest `cfg!(not(windows))` or just don't emit the warning. 

----

```
$ cargo clippy --version
0.0.143
```

**Comment by shssoichiro** [comment]
This is likely related to this issue: https://github.com/Manishearth/rust-clippy/issues/1877

Basically clippy is losing its mind on macros right now.

**Comment by kennytm** [comment]
Okay let's see if this will be fixed after rust-lang/rust#43179 is shipped.

**Comment by oli-obk** [comment]
Nope not fixed by it... Gotta figure out what else changed :/

**Comment by oli-obk** [comment]
It's not just clippy. Even rustc goes crazy:

```rust
while cfg!(unix) {} // better use a `loop` here, because "infinite loop" and "reasons" xD
```

**Comment by mcarton** [comment]
Maybe we should suggest a fix in rustc then

**Comment by oli-obk** [comment]
https://github.com/rust-lang/rust/issues/43268 done
